cmd/cmdutils_test.go

```diff
@@ -58,3 +68,133 @@ func Test_validateName(t *testing.T) {
 		})
 	}
 }
+
+func Test_validateStoragePath(t *testing.T) {
+
+	type args struct {
+		storagePath, componentName, applicationName string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Test Case 1",
+			args: args{
+				storagePath:     "/opt/app-root/src/storage/",
+				componentName:   "nodejs",
+				applicationName: "app",
+			},
+			wantErr: true,
+		},
+
+		{
+			name: "Test Case 2",
+			args: args{
+				storagePath:     "/opt/app-root/src/storage/test",
+				componentName:   "nodejs",
+				applicationName: "app",
+			},
+			wantErr: false,
+		},
+	}
+
+	pvcList := v1.PersistentVolumeClaimList{
+		Items: []v1.PersistentVolumeClaim{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "mystorage-app-pvc",
+					Labels: map[string]string{
+						"app.kubernetes.io/component-name": "nodejs",
+						"app.kubernetes.io/name":           "app",
+						"app.kubernetes.io/storage-name":   "mystorage",
+					},
+					Namespace: "myproject",
+				},
+			},
+		},
+	}
+
+	pvc := v1.PersistentVolumeClaim{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mystorage-app-pvc",
+			Labels: map[string]string{
+				"app.kubernetes.io/component-name": "nodejs",
+				"app.kubernetes.io/name":           "app",
+				"app.kubernetes.io/storage-name":   "mystorage",
+			},
+			Namespace: "myproject",
+		},
+	}
+
+	listOfDC := appsv1.DeploymentConfigList{
+		Items: []appsv1.DeploymentConfig{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "nodejs-app",
+					Namespace: "myproject",
+					Labels: map[string]string{
+						"app.kubernetes.io/component-name": "nodejs",
+						"app.kubernetes.io/component-type": "nodejs",
+						"app.kubernetes.io/name":           "app",
+					},
+				},
+				Spec: appsv1.DeploymentConfigSpec{
+					Template: &v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									VolumeMounts: []v1.VolumeMount{
+										{
+											MountPath: "/opt/app-root/src/storage/",
+											Name:      "mystorage-app-pvc-idrcg-volume",
+										},
+									},
+								},
+							},
+
+							Volumes: []v1.Volume{
+								{
+									Name: "mystorage-app-pvc-idrcg-volume",
+									VolumeSource: v1.VolumeSource{
+										PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+											ClaimName: "mystorage-app-pvc",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	client, fakeClientSet := occlient.FakeNew()
+	fakeClientSet.AppsClientset.PrependReactor("list", "deploymentconfigs", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, &listOfDC, nil
```

Comment:
Maybe here we can add a check for the selectors https://github.com/redhat-developer/odo/blob/3f11739e4c29583844222dc2f904dd0d982c8f33/pkg/url/url_test.go#L229